### ISSUE TITLE
Compatibility with 6.x version

### ISSUE THREAD

**Issue by Sneijky** [codingchili/kbn-authentication-plugin#27]
Hi,

I tried using your plugin on kibana 6.2.2. 
I followed all the instructions and after installing the plugin and running kibana, kibana runs normally and the plugin doesn't show. 

Looking at kibana console, I think the plugin isn't begin executed even though is installed.

kibana console:
![image](https://user-images.githubusercontent.com/38161593/40327979-af52a124-5d3c-11e8-9d8b-c1502877f823.png)

plugin list:
![image](https://user-images.githubusercontent.com/38161593/40328025-d1939c16-5d3c-11e8-90f5-6a1e2bae8487.png)

Have you ever tried to run the plugin in version 6.x or have a fix for the problem? 
This plugin fits my needs pretty well and would be amazing to have it working.

Thanks in Advance and keep the amazing work.

**Comment by codingchili** [comment]
Hello! 

The most recent version this plugin has been verified to work with is 5.6.2. I will make an effort to make sure it is compatible with 6+ as well. Expected within 1-2 weeks.

Thanks for submitting an issue.

**Comment by codingchili** [comment]
Update: I reproduced this in kibana/elasticsearch 6.2.3. The plugin installs fine but is never loaded. Nothing in the logs! I tried to create a new plugin to compare with. Problem is i can't build kibana because the version of yarn i just downloaded is broken, so the plugin-generator does not work.

I will at least try to update some dependencies for now and see if tests pass..

**Comment by codingchili** [comment]
Part of the template for plugins on 6.3

https://github.com/elastic/kibana/blob/master/packages/kbn-plugin-generator/sao_template/template/index.js

Added the config function, still not loading the plugin. 
Updated some dependencies which resolved the two existing vulnerable package warnings.

**Comment by codingchili** [comment]
Downloaded nightly yarn 1.8.0-20180601.1611 and it builds for 5 minutes before failing. 
`The engine "node" is incompatible with this module. Expected version "8.11.0"`

Using node 8.11.2.. trying a downgrade.

**Comment by codingchili** [comment]
Managed to build the plugin generator with 8.11.0, the generated plugin looks nearly identical to ours. Save for a 'name' field in the Kibana.plugin initializer. Adding the name and making one final attempt at booting it up before I need a break.

**Comment by codingchili** [comment]
It appears there is an issue loading native libraries (what a surprise, again and again). The argon2-ffi library works fine on my 64-bit 8.11.0 node but fails on the node version that ships with kibana, which is a 64-bit node on version 6.12.2. I had to use a small reproducer as this error does NOT appear when loading the library from a kibana plugin.. 

```
Loading argon2-ffi..
C:\Program Files\Kibana\node\node_modules\bindings\bindings.js:88
        throw e
        ^

Error: A dynamic link library (DLL) initialization routine failed.
\\?\C:\Program Files\Kibana\node\node_modules\ref-napi\build\Release\binding.node
    at Error (native)
    at Object.Module._extensions..node (module.js:597:18)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at bindings (C:\Program Files\Kibana\node\node_modules\bindings\bindings.js:81:44)
    at Object.<anonymous> (C:\Program Files\Kibana\node\node_modules\ref-napi\lib\ref.js:7:47)
    at Module._compile (module.js:570:32)
```

Definitely DONE for today with this. Grr.

**Comment by codingchili** [comment]
Appears that I closed this issue by accident.

Opened an issue here:
https://github.com/cjlarose/argon2-ffi/issues/27

**Comment by codingchili** [comment]
No response yet. Since the issue only affects 6.12.2  it would suggest an extreme peculiarity is at hand. More likely to change to non-ffi version of argon, which is at least 25x more popular. It is however not stable (?) because major is < 1.. however, we are in js-land and might have to go a bit wild western here.

We are moving to latest ES 6.3+ versions in ethereum-ingest and excelastic and it would be prudent to keep this fresh as well.

**Comment by codingchili** [comment]
Kibana 6.3.2 has moved to node 8.12, the plugin is tested to work with this version. There will be a new release with some minor style changes to better fit v6. No fix will be provided for prior versions.

**Comment by codingchili** [comment]
Released as 1.1.0, closing this issue.
